### PR TITLE
Base64 data issue/v8

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:  https://redmine.openinfosecfoundation.org/issues/6135

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1292

Previous PR: https://github.com/OISF/suricata/pull/9159

Changes since v7:
- Fix clang formatting

### CODE REVIEW

File: src/util-base64.c [hunk 1]
```diff
@@ -156,11 +156,26 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
...
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            return ecode;
+        }
+        /* if the destination size is not at least 3 Bytes long, it'll give a dynamic
+         * buffer overflow while decoding, so, return and let the caller take care of the
+         * remaining bytes to be decoded which should always be < 4 at this stage */
+        if (dest_size - *decoded_bytes < 3)
+            return BASE64_ECODE_OK;
```
Comment (victorjulien) [inline src/util-base64.c:173]: is this not an error?

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 592373632 | 519668040 | 87.73% | 

Pipeline 15021

**Comment by victorjulien** [comment]
```
  CC       util-base64.o
util-base64.c:177:9: warning: Value stored to 'sp' is never read [deadcode.DeadStores]
        sp = 0;
        ^    ~
1 warning generated.
```
scan-build warning ^

**Comment by victorjulien** [comment]
SV PR needs rebase?

File: src/util-base64.c (hunk 1 above)
Comment (inashivb) [reply src/util-base64.c:173]: My idea here was that if 2 bytes are permissible in dest_size, we return to the caller and let it take care of those two bytes in the next iteration/buffer like we do with the remaining bytes in MIME decoder. But, I certainly missed that this part is not called by MIME at all and the base64 decode keyword has no buffering mechanism.
So, yeah, this should probably be an error indeed.

**Pull request closed without merging**